src/rust/catloop/futures/connect.rs

```diff
@@ -279,9 +302,19 @@ fn connect_ack_received(
             duplex_pipe,
         };
     } else {
-        // Connection timeout, retry.
-        DuplexPipe::drop(&self_.catmem, qt_rx)?;
-        self_.state = ClientState::InitiateConnectRequest;
+        if attempt > MAX_CONNECT_ATTEMPTS {
```

Comment:
This name is a bit confusing. It's unclear at a glance whether it corresponds to `InitiateConnectRequest` or `ConnectAckReceived`. Is there a way to simplify this? May be naming the constant appropriately? (e.g. MAX_CONNECT_ACK_RECEIPT_ATTEMPTS / MAX_ACK_RECEIVED_ATTEMPTS etc.)

Reply:
Sure.